### DISCUSSION TITLE
report converters for kernel-motivated static analysis tools

### DISCUSSION THREAD

**Discussion by bulwahn** [Ericsson/codechecker#2938] [Ideas]
I am working in the LF ELISA Project, https://elisa.tech, and I am interested in using CodeChecker to track analyzer findings of Linux kernel-related tools on the kernel code repository. I am working together with others in the LF ELISA Project and working with students on the implementation of CodeChecker extensions.

Beyond the 'standard' static analysis tools, e.g., clang-tidy, clang-analyzer, the kernel community has further developed a set of static analysis tools, i.e., sparse, coccinelle, smatch. These tools mainly originated from the needs of the kernel community, but are applicable to other C code bases with a bit of additional work by other projects. More information on the tools is available here: https://www.kernel.org/doc/html/latest/dev-tools/index.html.

The overall goal here is to set up a productive environment for the kernel community. As the analysis tools are already integrated in a well configured and maintained way in the kernel build scripts, we do not need to invoke them with the CodeChecker analyzer commands. We can simply have setup scripts, that invoke the kernel build scripts, record the command line output of the tools and then pass this output to suitable report converters for those static analysis tools.

We would like to implement and contribute these report converters for sparse, coccinelle and smatch. Pull requests with the implementations for the report converters will follow.

The currently most promising work in progress is here:

- for coccinelle by Jay Rayput (mentorship from LF CommunityBridge, funded by LF ELISA Project):
  - https://github.com/Ericsson/codechecker/pull/2949 (merged)
  - https://github.com/Ericsson/codechecker/pull/2955 (merged)

- for sparse by Argert Boja (working student, funded by BMW AG):
  - https://github.com/Ericsson/codechecker/pull/2950 (work on open review comments required)

- for smatch by Jay Rayput and Sudip Mukherjee (ups, duplicated work, great minds think alike..):
  - https://github.com/Ericsson/codechecker/pull/2967 (obsolete, closed)
  - https://github.com/Ericsson/codechecker/pull/2968 (merged)
  - Still pending: waiting for smatch extension to be in smatch public git; then we provide documentation here.

- for kerneldoc warnings, assigned to Jay Rayput.
  - https://github.com/Ericsson/codechecker/pull/2981 (merged)

- for Python Sphinx Documentation warnings, assigned to Argert Boja (still investigating, not yet properly started)

**Comment by gyorb** [comment]
Hi @bulwahn,

Great news! Let us know if you have any questions about the converters or CodeChecker in general.
We are happy to help!

**Comment by bulwahn** [comment]
Just a quick update after we got the first two pull requests in shape, we will continue with parsing smatch output and with parsing Sphinx-build output (that is used for kernel documentation).
